ADAL/tests/ADAcquireTokenTests.m

```diff
@@ -1650,4 +1666,45 @@ - (void)testResilencyTokenDeletion
     XCTAssertEqual(allItems.count, 0);
 }
 
+- (void)testAllowSilentRequestParameters
+{
+    ADAuthenticationContext* context = [self getTestAuthenticationContext];
+    ADRequestParameters *params = [[ADRequestParameters alloc] initWithAuthority:context.authority
+                                                                        resource:TEST_RESOURCE
+                                                                        clientId:TEST_CLIENT_ID
+                                                                     redirectUri:TEST_REDIRECT_URL.absoluteString
+                                                                      identifier:[ADUserIdentifier identifierWithId:TEST_USER_ID]
+                                                                      tokenCache:context.tokenCacheStore
+                                                                extendedLifetime:NO
+                                                                   correlationId:nil
+                                                              telemetryRequestId:nil];
+    ADAuthenticationRequest* req = [ADAuthenticationRequest requestWithContext:context requestParams:params error:nil];
+    [req setSilent:YES];
+    [req setAllowSilentRequests:YES];
+    
+    
+    // Following we add a mock response and specify the request url we expect (it must include login_hint)
+    ADTestURLResponse* response = [ADTestURLResponse requestURLString:@"https://login.windows.net/contoso.com/oauth2/authorize?client_id=c3c7f5e5-7153-44d4-90e6-329686d48d76&resource=resource&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&nux=1&response_type=code&login_hint=eric_cartman%40contoso.com&x-client-Ver=" ADAL_VERSION_STRING
+                                                    responseURLString:@"https://idontmatter.com"
+                                                         responseCode:401
+                                                     httpHeaderFields:@{}
+                                                     dictionaryAsJSON:@{}];
+    [ADTestURLConnection addResponse:response];
+    
+    // We send the actual silent network request
+    [req requestToken:^(ADAuthenticationResult *result)
+     {
+         // if request url is not expected,
+         // our network mock will fail the unit test and it won't hit here
+         XCTAssertNotNil(result);
```

Comment:
this is a nit, and feel free to ignore in the context of this review, but usually best practice unit testing discourages asserting on things you don't actually care about in the context of this specific unit test. In other words, if you have other UTs that assert that result is not null, you don't need to also assert that here. Assert only the things you are actually testing in this test. That helps keep the UTs less brittle -- if a fundamental assertion changes you don't have to fix 100 tests.